aiida_workgraph/engine/utils.py

```diff
@@ -152,11 +153,19 @@ def prepare_for_shell_task(task: dict, kwargs: dict) -> dict:
         if key.startswith("nodes."):
             nodes[key[6:]] = kwargs.pop(key)
     metadata.update({"call_link_label": task["name"]})
+
+    arguments = kwargs.pop("arguments", [])
+    # Split on whitespace if arguments given as `str`
+    if isinstance(arguments, str):
+        arguments = shlex.split(arguments)
+    else:
+        lang.type_check(arguments, list, allow_none=True)
+
```

Comment:
Hi @GeigerJ2 , thanks for the work, the idea is to call the new `prepare_shell_job_inputs` (from aida-shell) inside this `prepare_for_shell_task` function, to replace lines 142-150 (prepare code etc), 157-162 (slit argument). Thus we don't need to manually update this function if aiida-shell updates.

Reply:
Ah, OK, I see. Yeah, I was also wondering why we duplicate some code from that function here, but thought there might be a reason to handle some things differently in WorkGraph.

Reply:
Wasn't aware of the relevant [PR](sphuber/aiida-shell#103) in `aiida-shell`. Now using the `prepare_shell_job_inputs` helper function, and just updating the `kwargs` `nodes` and `metadata` as required.